client/app/bundles/course/assessment/submission/containers/ProgrammingImportEditor.jsx

```diff
@@ -87,18 +104,24 @@ class VisibleProgrammingImportEditor extends Component {
           handleDeleteFile={this.handleDeleteFile}
           handleFileTabbing={index => this.setState({ displayFileIndex: index })}
           files={files}
+          viewHistory={viewHistory}
         />}
-        <FieldArray
-          name={`${answerId}[files_attributes]`}
-          component={VisibleProgrammingImportEditor.renderSelectProgrammingFileEditor}
-          {...{
-            readOnly,
-            question,
-            displayFileIndex,
-            language: parseLanguages(question.language),
-          }}
-        />
-        {readOnly ?
+        {viewHistory ?
+          this.renderProgrammingHistoryEditor(answers[answerId])
+          :
+          <FieldArray
+            name={`${answerId}[files_attributes]`}
+            component={VisibleProgrammingImportEditor.renderSelectProgrammingFileEditor}
+            {...{
+              readOnly,
+              question,
+              displayFileIndex,
+              viewHistory,
+              language: parseLanguages(question.language),
+            }}
+          />
+        }
+        {readOnly || viewHistory ?
```

Comment:
Just a random thought, I think no need to act on it for now. If now we have `readOnly` and `viewHistory`, there are the following states (in order of `readOnly, viewHistory`):
 1. `false, false`: actively doing assessment, can edit.
 1. `true, false`: view latest answer after assessment submitted
 1. `true, true`: viewing past answers

One question I have is for states 2 and 3, should they be more similar - any thoughts? Currently the difference is only whether the test cases are collapsed or not.

Reply:
For 2 & 3, there are differences elsewhere as well. Such as for multiple files submission, for state 2, the files are displayed one after another, whereas for state 3, they are displayed in tabs.
Either way, i don't think `readOnly` and `viewHistory` can be combined or simplified another way. They are kinda two disjunctive sets.

Reply:
For state 2 in multiple file submission, files are displayed one after another once the answer is submitted? Hmm should those be tabs too, to make it more consistent with the viewing of answers?

Reply:
Perhaps, the only thing of concern would be that it might not be clear when a 'tab' has a comment, that was why i left it like that in the original implementation. 

For viewing of past answers, it was put in tabs mainly so that we don't have a gigantic vertical scroll. Also, it is might not be likely that we will see many comments on different files in past answers.